otherlibs/systhreads/st_stubs.c

```diff
@@ -208,11 +222,10 @@ static void caml_thread_leave_blocking_section(void)
 {
   /* Wait until the runtime is free */
   st_masterlock_acquire(&Thread_main_lock);
-  /* Update Active_thread to point to the thread descriptor corresponding to
-     the thread currently executing */
-  Active_thread = st_tls_get(caml_thread_key);
-  /* Restore the runtime state from the curr_thread descriptor */
-  caml_thread_restore_runtime_state();
```

Comment:
I'm not fond of this API change because we lose the clear bracketing between `save_runtime_state` and `restore_runtime_state`, I think it makes the code harder to read.

Maybe another API coudl like this:
- here we only need to call `restore_runtime_state()` (no argument)
- `restore_runtime_state` is in charge of updating `Active_thread` and then calling `restore_runtime_state_from` on Active_thread
- `set_active` calls `restore_runtime_state_from` as well

Reply:
Fixed, please have a look

Reply:
I'm still a bit ambivalent on this part, but I don't quite remember the previous iteration.
I think the current proposal is fine however, do you have any update on your comments @gasche ?

Reply:
> I don't quite remember the previous iteration.

In case the old one is no longer visible from github, I think the new version addresses @gasche's criticism and the result is better (without losing what I intended to do).